src/Raven.Server/Documents/ResourceCache.cs

```diff
@@ -165,20 +165,10 @@ public IDisposable RemoveLockAndReturn(string databaseName, Action<TResource> on
 
                 task.IgnoreUnobservedExceptions();
 
-                bool found = false;
-                while (found == false)
```

Comment:
I was trying to find the reason why we had it handled this way. I found this PR: https://github.com/ravendb/ravendb/pull/4997 but I'm not sure how that was meant to fix the issue https://issues.hibernatingrhinos.com/issue/RavenDB-10054.

@karmeli87 effectively your changes gets the code back to previous version? Can you think about any possible problems now?

Reply:
Not exactly... in the PR you showed, we still added that faulted task (line 140)
TBH, I can't see a reason why to add that task if the cache is empty..

Reply:
What race we want to solve here?

The only one that I can think of is adding a database before the `_caseInsensitive.TryGetValue` is executed.. 
Which would resolve into either throwing because the database is loading, or removing the database after it was loaded.. both are expected and ok.. I think.

Reply:
Consider removing a database that is unloaded, and at the same time trying to load it.
Because it was unloaded, you didn't take the lock, and then you have a race.